Actions/CheckForUpdates/yamlclass.ps1

```diff
@@ -194,4 +233,99 @@ class Yaml {
+                if ($jobsWithDependency) {
+                    Write-Host "  - Jobs with dependency: $($jobsWithDependency -join ', ')"
+                }
+                else {
+                    Write-Host "  - No jobs with dependency on this"
+                }
+                $result += @(@{ "Name" = $customJob; "Content" = @($this.Get("jobs:/$($customJob):").content); "NeedsThis" = $jobsWithDependency })
+            }
+        }
+        return $result
+    }
+
+    # Add jobs to Yaml and update Needs section from native jobs which needs this custom Job
+    # $customJobs is an array of hashtables with Name, Content and NeedsThis
+    # Example:
+    # $customJobs = @(@{ "Name" = "CustomJob1"; "Content" = @("  - pwsh","  -File Build1"); "NeedsThis" = @("Initialization", "Build") })
+    # AddCustomJobsToYaml($customJobs)
+    # The function will add the job CustomJob1 to the Yaml file and update the Needs section of Initialization and Build
+    # The function will not add the job CustomJob1 if it already exists
+    [void] AddCustomJobsToYaml([hashtable[]] $customJobs) {
+        $existingJobs = $this.GetNextLevel('jobs:/').Trim(':')
+        Write-Host "Adding New Jobs"
+        foreach($customJob in $customJobs) {
+            Write-Host "$($customJob.Name) has dependencies from $($customJob.NeedsThis -join ',')"
+            foreach($needsthis in $customJob.NeedsThis) {
+                if ($needsthis -eq 'Build') {
+                    $existingJobs | Where-Object { $_ -like 'Build*'} | ForEach-Object {
+                        # Add dependency to all build jobs
+                        $this.Replace("jobs:/$($_):/needs:","needs: [ $(@($this.GetPropertyArray("jobs:/$($_):/needs:"))+@($customJob.Name) -join ', ') ]")
+                    }
+                }
+                elseif ($existingJobs -contains $needsthis) {
+                    # Add dependency to job
+                    $needs = @(@($this.GetPropertyArray("jobs:/$($needsthis):/needs:"))+@($customJob.Name) | Where-Object { $_ } | Select-Object -Unique) -join ', '
+                    $this.Replace("jobs:/$($needsthis):/needs:","needs: [ $needs ]")
+                }
+            }
+            if ($existingJobs -contains $customJob.Name) {
+                Write-Host "Job $($customJob.Name) already exists"
+                continue
+            }
+            $this.content += @('') + @($customJob.content | ForEach-Object { "  $_" })
+        }
+    }
+
+    static [void] ApplyCustomizations([ref] $srcContent, [string] $yamlFile) {
+        $srcYaml = [Yaml]::new($srcContent.Value.Split("`n"))
+        try {
+            $yaml = [Yaml]::Load($yamlFile)
+        }
+        catch {
+            Write-Host "Unable to read YAML file $yamlFile. Skipping custom jobs."
+            return
+        }
+
+        # Locate custom jobs in destination YAML
+        Write-Host "Apply custom jobs"
+        $customJobs = @($yaml.GetCustomJobsFromYaml('CustomJob*'))

```

Comment:
Minor: convention

Why does the name have to match "CustomJob*"? Aren't all jobs that are not in the original template custom jobs?
With the exception of the additional build jobs (if any), "Build1", "Build2", etc.

Reply:
If we don't keep the CustomJob prefix, chances are that people will name their custom jobs like jobs we might add or rename in the future.
Prefixing makes this very clear and has no disadvantages - you can always add a display name to your job.